runtime/compiler/optimizer/abstractinterpreter/AbsInterpreter.cpp

```diff
@@ -0,0 +1,3072 @@
+   {
+   if (type == TR::NoType)
+      return;
+
+   TR::AbsStackMachineState* state = getState();
+   if (type.isDouble() || type.isInt64())
+      state->pop();
+
+   TR::AbsValue* value = state->pop();
+   TR_ASSERT_FATAL(type == TR::Int16 || type == TR::Int8 ? value->getDataType() == TR::Int32 : value->getDataType() == type, "Unexpected type");
+   }
+
+void TR::AbsBlockInterpreter::constant(int32_t i)
+   {
+   TR::AbsStackMachineState* state = getState();
+   TR::AbsValue* value = createIntConst(i);
+   state->push(value);
+   }
+
+void TR::AbsBlockInterpreter::constant(int64_t l)
+   {
+   TR::AbsStackMachineState* state = getState();
+   TR::AbsValue* value1 = createLongConst(l);
+   TR::AbsValue* value2 = createTopLong();
+   state->push(value1);
+   state->push(value2);
+   }
+
+void TR::AbsBlockInterpreter::constant(float f)
+   {
+   TR::AbsStackMachineState* state = getState();
+   TR::AbsValue* floatConst = createTopFloat();
+   state->push(floatConst);
+   }
+
+void TR::AbsBlockInterpreter::constant(double d)
+   {
+   TR::AbsStackMachineState* state = getState();
+   TR::AbsValue *value1 = createTopDouble();
+   TR::AbsValue *value2 = createTopDouble();
+   state->push(value1);
+   state->push(value2);
+   }
+
+void TR::AbsBlockInterpreter::constantNull()
+   {
+   TR::AbsStackMachineState* state = getState();
+   TR::AbsValue* value = createNullObject();
+   state->push(value);
+   }
+
+void TR::AbsBlockInterpreter::ldc(bool wide)
+   {
+   TR::AbsStackMachineState* state = getState();
+
+   int32_t cpIndex = wide ? _bci.next2Bytes() : _bci.nextByte();
+   TR::DataType type = _callerMethod->getLDCType(cpIndex);
+
+   switch (type)
+      {
+      case TR::Int32:
+         {
+         int32_t intVal = _callerMethod->intConstant(cpIndex);
+         constant((int32_t)intVal);
+         break;
+         }
+      case TR::Int64:
+         {
+         int64_t longVal = _callerMethod->longConstant(cpIndex);
+         constant((int64_t)longVal);
+         break;
+         }
+      case TR::Double:
+         {
+         constant((double)0); //the value does not matter here since doubles are modeled as TOP
+         break;
+         }
+      case TR::Float:
+         {
+         constant((float)0); //same for float
+         break;
+         }
+      case TR::Address:
+         {
+         bool isString = _callerMethod->isStringConstant(cpIndex);
+         if (isString)
+            {
+            TR::SymbolReference *symbolReference = comp()->getSymRefTab()->findOrCreateStringSymbol(_callerMethodSymbol, cpIndex);
+            if (symbolReference->isUnresolved())
+               {
+               state->push(createTopObject());
+               }
+            else  //Resolved
+               {
+               TR::AbsValue *stringVal = createStringObject(symbolReference, TR_yes);
+               state->push(stringVal);
+               }
+            }
+         else  //Class
```

Comment:
Since BenefitInliner currently doesn't support method type, method handle and condy, I changed this part to proceed as class constant only if `_callerMethod->isClassConstant(cpIndex)` is true. For any other type, we will just push a top into the stack now.